### PR TITLE
test_url_query: Expand tests

### PR DESCRIPTION
**Pull request by behnam**
Expanding URL-query tests to cover more behaviors around special chars
when parsing, setting, re-setting and getting values.

Bug revealed are marked with `FIXME` to address separately.

### CODE REVIEW

**Comment by behnam** [comment]
Looks like we have a conflict here: `mypy` is enforced in CI and I need to use py36-only syntax for type hinting to make it happy, but then it fails on py35.

I think we should limit mypy to py>=36. Let me send a separate PR for that.

File: tests/test_url_query.py [hunk 1]
```diff
@@ -1,37 +1,82 @@
...
-
-
-def test_query_spaces():
-    url = URL('http://example.com?a+b=c+d')
-    assert url.query == MultiDict({'a b': 'c d'})
-
-
-def test_query_empty():
-    url = URL('http://example.com')
-    assert isinstance(url.query, MultiDictProxy)
```
Comment (asvetlov) [inline tests/test_url_query.py]: Please keep a check for type returned by `url.query`

**Review by asvetlov — DISMISSED** [review]

File: tests/test_url_query.py [hunk 2]
```diff
@@ -1,37 +1,83 @@
...
+    ),
+]  # type: List[Tuple[URL, MultiDict]]
+
+
+@pytest.mark.parametrize(
+    'original_url, expected_query',
+    URLS_WITH_BASIC_QUERY_VALUES,
+)
+def test_query_basic_parsing(original_url, expected_query):
+    assert isinstance(original_url.query, MultiDictProxy)
```
Comment (behnam) [inline tests/test_url_query.py]: Added here.

**Comment by behnam** [comment]
@asvetlov, I believe this is ready for review.

**Review by mjpieters — APPROVED** [review]
LGTM, and the original change request has been addressed.

**Comment by mjpieters** [comment]
As I can't push a minor fix to this PR, I'll re-create it and merge that instead.

**Pull request merged**